### PR TITLE
Added support for tuya sensor, ZG-227ZL and ZG-102ZL

### PR DESCRIPTION
**Pull request by 13717033460**
Hello, it has been resubmitted according to the new format of tuya。

### CODE REVIEW

**Comment by 13717033460** [comment]
Hello, it has been resubmitted according to the new format of tuya

File: devices/tuya.js [hunk 1]
```diff
@@ -3417,8 +3417,7 @@ module.exports = [
         ],
     },
     {
-        fingerprint: [{modelID: 'TS0601', manufacturerName: '_TZE200_qoy0ekbd'},
-            {modelID: 'TS0601', manufacturerName: '_TZE200_znbl8dj5'}],
+        fingerprint: [{modelID: 'TS0601', manufacturerName: '_TZE200_znbl8dj5'}],
```
Comment (Koenkk) [inline devices/tuya.js:3420]: `_TZE200_qoy0ekbd` and `_TZE200_znbl8dj5` are already specified here, is this the same device? (CX-0726)

Comment (13717033460) [reply devices/tuya.js:3420]: tuya developer account, device PID, screenshot
https://github.com/13717033460/screenshots/blob/main/ZG-227ZL.png
https://github.com/13717033460/screenshots/blob/main/ZG-227z.png
https://github.com/13717033460/screenshots/blob/main/ZG-102zl.png

File: devices/tuya.js [hunk 2]
```diff
@@ -3549,4 +3537,58 @@ module.exports = [
...
+        model: 'ZG-225Z',
+        vendor: 'TuYa',
+        description: 'Gas sensor',
+        fromZigbee: [tuya.fzDataPoints],
+        toZigbee: [tuya.tzDataPoints],
+        configure: tuya.configureMagicPacket,
+        exposes: [tuya.exposes.gasState(), tuya.exposes.gasValue().withUnit('%')],
+        meta: {
+            tuyaDatapoints: [
+                [1, 'gas_state', tuya.valueConverter.gasState],
```
Comment (Koenkk) [inline devices/tuya.js:3589]: It is better to make gas a boolean:
- `gas_state` -> `gas` 
- `tuya.valueConverter.gasState` -> `tuya.valueConverter.true0ElseFalse`
- `tuya.exposes.gasState()` -> `e.gas()`

Comment (13717033460) [reply devices/tuya.js:3589]: Thanks for your suggestion，i whill change it to boolean.

File: lib/tuya.js [hunk 3]
```diff
@@ -1170,6 +1170,13 @@ const tuyaExposes = {
     errorStatus: () => exposes.numeric('error_status', ea.STATE).withDescription('Error status'),
     scheduleAllDays: (access, format) => ['monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday', 'sunday']
         .map((day) => exposes.text(`schedule_${day}`, access).withDescription(`Schedule for ${day}, format: "${format}"`)),
+    temperatureUnit: () => exposes.enum('temperature_unit', ea.STATE_SET, ['celsius', 'fahrenheit']).withDescription('Temperature unit'),
+    temperatureCalibration: () => exposes.numeric('temperature_calibration', ea.STATE_SET).withValueMin(-2.0).withValueMax(2.0)
+        .withValueStep(0.1).withUnit('°C').withDescription('Temperature calibration'),
+    humidityCalibration: () => exposes.numeric('humidity_calibration', ea.STATE_SET).withValueMin(-30).withValueMax(30)
+        .withValueStep(1).withUnit('%').withDescription('Humidity calibration'),
+    gasState: () => exposes.enum('gas_state', ea.STATE, ['alarm', 'normal']).withDescription('Gas detection state'),
+    gasValue: () => exposes.numeric('gas_value', ea.STATE).withDescription('Detected gas'),
```
[the remark this answers is not in the document]
Comment (13717033460) [reply lib/tuya.js:1179]: gasValue:  gas concentration value, unit :ppm

**Comment by Koenkk** [comment]
Thanks!

**Pull request merged**